### PR TITLE
common: animation improvements

### PR DESCRIPTION
**Pull request by fjl**
Please see commit messages for change details.

### CODE REVIEW

**Comment by fjl** [comment]
Is there a reason why `NextFrame` and `Cell` are exported? They're only used by AnimationSystem and will crash if used incorrectly.

File: common/animation.go [hunk 1]
```diff
@@ -80,41 +94,39 @@ func (ac *AnimationComponent) NextFrame() {
...
-	delete := -1
-	for index, e := range a.entities {
-		if e.BasicEntity.ID() == basic.ID() {
-			delete = index
-			break
-		}
-	}
-	if delete >= 0 {
-		a.entities = append(a.entities[:delete], a.entities[delete+1:]...)
+	if a.entities != nil {
```
Comment (EtienneBruines) [inline common/animation.go]: If you add this, then you won't have to do these checks. 

```go
func (a *AnimationSystem) New(*ecs.World) {
    a.entities = make(map[ecs.BasicEntity]animationEntity)
}
```

**Comment by EtienneBruines** [comment]
Very nice, and thank you for your contribution!

The `master` branch changed a bit (just merged a PR), so you may want to update your local branch before doing any changes. 

I have reviewed your PR and it looks pretty nice. There was one thing I commented, maybe you want to have a look at that.

File: common/animation.go (hunk 1 above)
Comment (fjl) [inline common/animation.go]: That's true, but it also means I can't add entities before adding the system to World. It'd be nice to keep that working.

**Comment by fjl** [comment]
@EtienneBruines can you answer my question about Cell and NextFrame

**Comment by otraore** [comment]
@fjl In response to your question, I don't believe `Cell()` will crash the program if used incorrectly. It's just a reference to the current animation frame which can be useful . As for `NextFrame(`) it also just allows the user more flexibility. For example, they can stop the animation loop and manually trigger the frames. If you still think they either of them should be unexported, you can create an issue where we can discuss it.

With that said, On behalf of the Engo team I'd like to thank you for your contribution. 
LGTM.

**Pull request merged**